src/main/java/org/citopt/connde/web/rest/RestComponentFilterController.java

```diff
@@ -30,6 +34,39 @@
     @Autowired
     private SensorRepository sensorRepository;
 
+    @Autowired
+    private RuleActionRepository ruleActionRepository;
+
+    @Autowired
+    private RuleRepository ruleRepository;
+
+    /**
+     * Returns a list of rules that make use of a certain rule action.
+     *
+     * @param ruleActionId The id of the rule action for which using rules should be found
+     * @return A list of all rules that make use of the rule action
+     */
+    @GetMapping("/rules/by-ruleAction/{id}")
+    @ApiOperation(value = "Retrieves the rules which make use of a certain rule action and for which the user is authorized", produces = "application/hal+json")
+    @ApiResponses({@ApiResponse(code = 200, message = "Success")})
+    public ResponseEntity<List<Rule>> getRulesByRuleActionID(@PathVariable(value = "id") @ApiParam(value = "ID of the rule action", example = "5c97dc2583aeb6078c5ab672", required = true) String ruleActionId) {
+
+        RuleAction ruleAction = ruleActionRepository.findOne(ruleActionId);
+        List<Rule> rules = ruleRepository.findAll();

```

Comment:
Hmm. I have a question on this function. Am I supposed to filter these rules, so that I consider just the ones the user is authorized??

Reply:
No, not necessary, the call of `ruleRepository.findAll()` does this for you.
See [this line]( https://github.com/IPVS-AS/MBP/blob/da0bc7bace5f5b64cc9ac70e06a897eed15f82ee/src/main/java/org/citopt/connde/repository/RuleRepository.java#L50) for more details :-)